common/fixation_entry.go

```diff
@@ -0,0 +1,311 @@
+	// get the relevant store
+	store := prefix.NewStore(ctx.KVStore(vs.storeKey), types.KeyPrefix(types.EntryKey+fixationKey+index))
+	byteKey := types.KeyPrefix(createEntryKey(index, fixationKey, block))
+
+	// marshal the new entry data
+	b := vs.cdc.MustMarshal(entryData)
+
+	// get the entry from the store
+	entry := vs.getUnmarshaledEntryForBlock(ctx, fixationKey, index, block, types.DO_NOTHING)
+	if entry == nil {
+		return utils.LavaError(ctx, ctx.Logger(), "SetEntry_cant_find_entry", map[string]string{"fixationKey": fixationKey, "index": index, "block": strconv.FormatUint(block, 10)}, "can't set non-existent entry")
+	}
+
+	// update the entry's data
+	entry.Data = b
+
+	// marshal the entry
+	bz := vs.cdc.MustMarshal(entry)
+
+	// set the entry
+	store.Set(byteKey, bz)
+
+	return nil
+}
+
+// handleRefAction handles ref actions: increase refs, decrease refs or do nothing
+func handleRefAction(ctx sdk.Context, entry *types.Entry, refAction types.ReferenceAction) error {
+	// check if entry is nil
+	if entry == nil {
+		return utils.LavaError(ctx, ctx.Logger(), "handleRefAction_nil_entry", map[string]string{}, "can't handle reference action, entry is nil")
+	}
+
+	// handle the ref action
+	switch refAction {
+	case types.ADD_REFERENCE:
+		entry.References += 1
+	case types.SUB_REFERENCE:
+		if entry.GetReferences() > 0 {
+			entry.References -= 1
+		}
+	case types.DO_NOTHING:
+	}
+
+	return nil
+}
+
+// GetEntry gets the exact entry with index and block (block has to be precise). The user should pass an empty pointer of the desired type (e.g. package, subsciption, etc.).
+func (vs VersionedStore) GetEntry(ctx sdk.Context, fixationKey string, index string, block uint64, entryData codec.ProtoMarshaler, refAction types.ReferenceAction) bool {
+	// create entry key
+	entryKey := createEntryKey(index, fixationKey, block)
+
+	// get the relevant store
+	store := prefix.NewStore(ctx.KVStore(vs.storeKey), types.KeyPrefix(types.EntryKey+fixationKey+index))
+
+	// get the marshled entry
+	bz := store.Get(types.KeyPrefix(entryKey))
+
+	// couldn't find entry
+	if bz == nil {
+		return false
+	}
+
+	// unmarshal the entry
+	var entry types.Entry
+	vs.cdc.MustUnmarshal(bz, &entry)
+
+	// handle ref action
+	err := handleRefAction(ctx, &entry, refAction)
+	if err != nil {
+		return false
+	}
+
+	// unmarshal the entry's data
+	vs.cdc.MustUnmarshal(entry.GetData(), entryData)
+
+	return true
+}
+
+// GetStoreKey returns the versioned store's store key
+func (vs VersionedStore) GetStoreKey() sdk.StoreKey {
+	return vs.storeKey
+}
+
+// GetCdc returns the versioned store's codec
+func (vs VersionedStore) GetCdc() codec.BinaryCodec {
+	return vs.cdc
+}
+
+// getUnmarshaledEntryForBlock gets an entry by block. Block doesn't have to be precise, it gets the closest entry version
+func (vs VersionedStore) getUnmarshaledEntryForBlock(ctx sdk.Context, fixationKey string, index string, block uint64, refAction types.ReferenceAction) *types.Entry {
```

Comment:
I think this name fits (corresponds to other functions wrote in the past). Also, I don't think we need a SetRawEntry since it's supposed to be private (setRawEntry) and there is no reason for the caller to use it